### PR TITLE
Spin until future complete

### PR DESCRIPTION
**Pull request by jacquelinekay**
Connects to ros2/rclcpp#111

^ the card for that issue on Waffle disappeared mysteriously, even though I believe I've linked the card correctly. Argh.

### CODE REVIEW

**Comment by esteve** [comment]
+1

File: test_rclcpp/test/test_spin.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    };
+  auto timer = node->create_wall_timer(std::chrono::milliseconds(500), callback);
+
+  ASSERT_EQ(rclcpp::spin_until_future_complete(node, future),
+    rclcpp::executors::FutureReturnCode::SUCCESS);
+  EXPECT_EQ(future.get(), true);
+}
+
+TEST(CLASSNAME(test_spin, RMW_IMPLEMENTATION), spin_until_future_complete_timeout) {
+  auto node = rclcpp::Node::make_shared("test_spin_until_future_complete");
```
Comment (dirk-thomas) [inline test_rclcpp/test/test_spin.cpp]: The node name should match the test name.

Same below.

**Comment by jacquelinekay** [comment]
CI:
http://ci.ros2.org/job/ros2_batch_ci_linux/361/
http://ci.ros2.org/job/ros2_batch_ci_osx/271/
http://ci.ros2.org/job/ros2_batch_ci_windows/407/

The new test runs for all 3 rmw implementations:
http://ci.ros2.org/job/ros2_batch_ci_linux/361/testReport/%28root%29/test_spin__rmw_connext_cpp/
http://ci.ros2.org/job/ros2_batch_ci_linux/361/testReport/%28root%29/test_spin__rmw_connext_dynamic_cpp/
http://ci.ros2.org/job/ros2_batch_ci_linux/361/testReport/%28root%29/test_spin__rmw_opensplice_cpp/

**Comment by dirk-thomas** [comment]
+1

**Pull request merged**